### PR TITLE
misc updates

### PR DESCRIPTION
**Pull request by fredrikekre**
- cache celldofs in the iterator
- show method for grid and fevalues to increase quality of life in notebook/repl: This

```julia
julia> grid = generate_grid(Hexahedron, (20, 20, 20))
JuAFEM.Grid{3,8,Float64,6} with 8000 Hexahedron cells and 9261 nodes
```
instead of this:
```jl
julia> grid = generate_grid(Hexahedron, (20, 20, 20))
JuAFEM.Grid{3,8,Float64,6}(JuAFEM.Cell{3,8,6}[JuAFEM.Cell{3,8,6}((1, 2, 23, 22, 442, 443,
464, 463)), JuAFEM.Cell{3,8,6}((2, 3, 24, 23, 443, 444, 465, 464)), JuAFEM.Cell{3,8,6}((3,
4, 25, 24, 444, 445, 466, 465)), JuAFEM.Cell{3,8,6}((4, 5, 26, 25, 445, 446, 467, 466)),
...
```

### CODE REVIEW

File: src/FEValues/common_values.jl [hunk 1]
```diff
@@ -344,3 +344,7 @@ function spatial_coordinate{dim, T}(fe_v::Values{dim}, q_point::Int, x::Abstract
     end
     return vec
 end
+
+function Base.show(io::IO, fe_v::Values)
+    println(io, "$(typeof(fe_v)) with $(getnbasefunctions(fe_v)) base functions")
```
Comment (fredrikekre) [inline src/FEValues/common_values.jl]: dont really know whats interesting to show here...

File: src/iterators.jl [hunk 2]
```diff
@@ -71,6 +73,10 @@ function reinit!{dim, N}(ci::CellIterator{dim, N}, i::Int)
         ci.nodes[j] = nodeid
         ci.coords[j] = ci.grid.nodes[nodeid].x
     end
+    if isdefined(ci, :dh) # update celldofs
+        resize!(ci.celldofs, ndofs_per_cell(ci.dh))
```
Comment (fredrikekre) [inline src/iterators.jl]: perhaps we can put it to the correct size from the beginning?

File: src/Dofs/DirichletBoundaryConditions.jl [hunk 3]
```diff
@@ -1,9 +1,9 @@
-export DirichletBoundaryConditions, update!, apply!, apply_zero!, add!
+export DirichletBoundaryConditions, update!, apply!, apply_zero!, add!, free_dofs
 
 """
     DirichletBoundaryConditions
 
-A dirichlet boundary conditions is a a boundary where a dof is fixed to take a certain value.
+A Dirichlet boundary condition is a boundary where the function is fixed to take a certain value.
```
Comment (KristofferC) [inline src/Dofs/DirichletBoundaryConditions.jl]: the solution?

File: src/FEValues/common_values.jl (hunk 1 above)
Comment (KristofferC) [inline src/FEValues/common_values.jl]: perhaps the reference shape?

File: src/iterators.jl (hunk 2 above)
Comment (KristofferC) [inline src/iterators.jl]: Yes, why not just `zeros(Int, ndofs_per_cell(dh))` in the constructor?

Comment (fredrikekre) [inline src/iterators.jl]: that works as long as we have the same dofs in all elements :)

File: src/FEValues/common_values.jl (hunk 1 above)
Comment (fredrikekre) [inline src/FEValues/common_values.jl]: sure

Comment (fredrikekre) [inline src/FEValues/common_values.jl]: Ideally we should show the interpolation but thats gone in fevalues now. Any way to retrieve it?

File: src/iterators.jl (hunk 2 above)
Comment (KristofferC) [inline src/iterators.jl]: I guess... Do we have any thoughts to not have that. Even with different interpolation of fields there would still be the same number of dofs per element.

File: src/FEValues/common_values.jl (hunk 1 above)
Comment (KristofferC) [inline src/FEValues/common_values.jl]: store it in a string or something..

Comment (fredrikekre) [inline src/FEValues/common_values.jl]: reference shape is actually shown from the type:
```julia
julia> CellScalarValues(qr, ip)
JuAFEM.CellScalarValues{2,Float64,JuAFEM.RefCube} with 4 base functions
```

Comment (fredrikekre) [inline src/FEValues/common_values.jl]: Something like

```julia
julia> CellScalarValues(qr, ip)
JuAFEM.CellScalarValues{2,Float64,JuAFEM.RefCube} with 4 shape functions and 4 quadrature points
```

File: src/iterators.jl (hunk 2 above)
Comment (KristofferC) [inline src/iterators.jl]: Yeah.. but everything we have already breaks for that :P

File: examples/hyperelasticity.ipynb [hunk 4]
```diff
@@ -427,6 +425,7 @@
   }
  ],
  "metadata": {
+  "anaconda-cloud": {},
```
Comment (fredrikekre) [inline examples/hyperelasticity.ipynb]: Okej.

**Comment by fredrikekre** [comment]
good to go?

**Pull request merged**